tree/dataframe/src/RNTupleDS.cxx

```diff
@@ -337,8 +340,14 @@ void RNTupleDS::AddField(const RNTupleDescriptor &desc, std::string_view colName
          valueField =
             std::make_unique<ROOT::Experimental::RArrayAsRVecField>("", std::move(valueField), fieldInfo.fNRepetitions);
       } else {
-         // Actual ROOT::RVec
-         valueField = std::make_unique<ROOT::Experimental::RRVecField>("", std::move(valueField));
+         // Actual collection. A std::vector or ROOT::RVec gets added as a ROOT::RVec. All other collection types keep
+         // their original type.
+         if (convertToRVec) {
+            valueField = std::make_unique<ROOT::Experimental::RRVecField>("", std::move(valueField));
+         } else {
+            auto outerFieldType = desc.GetFieldDescriptor(fieldInfo.fFieldId).GetTypeName();
+            valueField = RFieldBase::Create("", outerFieldType).Unwrap();
```

Comment:
I think this may break the projection of inner collection members as individual collections. So, if you have
an `std::set<Point>` with `struct Point { float x,y; };`, are the columns for `x`, and `y` correctly exposed as `std::set<float>`?

Reply:
I think we cannot expose the individual columns as `std::set`: Even if the complete `Point` is unique in the set, the individual `x` and `y` may not. I think we have to expose them as sequence collection.